cli/command/container/run.go

```diff
@@ -184,7 +184,19 @@ func runContainer(dockerCli *command.DockerCli, opts *runOptions, copts *contain
 		}
 	}
 
-	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, copts.autoRemove)
+	if attach {
```

Comment:
The block above this is already `if attach {...}` could we merge these two blocks?